xdsl/interpreters/experimental/wgsl_printer.py

```diff
@@ -0,0 +1,248 @@
+            f"""
+        let {name_hint}: u32 = local_invocation_id.{dim};"""
+        )
+
+    @print.register
+    def _(self, op: gpu.GridDimOp, out_stream: IO[str]):
+        dim = str(op.dimension.value.param).replace('"', "")
+        name_hint = self.wgsl_name(op.result)
+        out_stream.write(
+            f"""
+        let {name_hint}: u32 = num_workgroups.{dim};"""
+        )
+
+    @print.register
+    def _(self, op: gpu.BlockDimOp, out_stream: IO[str]):
+        dim = str(op.dimension.value.param).replace('"', "")
+        name_hint = self.wgsl_name(op.result)
+        out_stream.write(
+            f"""
+        let {name_hint}: u32 = local_invocation_id.{dim};"""
+        )
+
+    @print.register
+    def _(self, op: gpu.GlobalIdOp, out_stream: IO[str]):
+        dim = str(op.dimension.value.param).replace('"', "")
+        name_hint = self.wgsl_name(op.result)
+        out_stream.write(
+            f"""
+        let {name_hint}: u32 = global_invocation_id.{dim};"""
+        )
+
+    @print.register
+    def _(self, op: memref.Load, out_stream: IO[str]):
+        memref_type = cast(MemRefType[Attribute], op.memref.type)
+        memref_dimension = memref_type.get_num_dims()
+        memref_size = memref_type.get_shape()
+        load_ref = self.wgsl_name(op.memref)
+        name_hint = self.wgsl_name(op.res)
+        indices = [self.wgsl_name(i) for i in op.indices]
+        index_value = self.calculate_index(memref_dimension, memref_size, indices)
+        out_stream.write(
+            f"""
+        let {name_hint} = {load_ref}[{index_value}];"""
+        )
+
+    @print.register
+    def _(self, op: memref.Store, out_stream: IO[str]):
+        memref_type = cast(MemRefType[Attribute], op.memref.type)
+        memref_dimension = memref_type.get_num_dims()
+        memref_size = memref_type.get_shape()
+        value = self.wgsl_name(op.value)
+        store_ref = self.wgsl_name(op.memref)
+        indices = [self.wgsl_name(i) for i in op.indices]
+        index_value = self.calculate_index(memref_dimension, memref_size, indices)
+        out_stream.write(
+            f"""
+        {store_ref}[{index_value}] = {value};"""
+        )
+
+    def calculate_index(
+        self, memref_dimension: int, memref_size: tuple[int], indices: list[str]
+    ):
+        index_values: list[str] = []
+        for i in range(memref_dimension):
+            product_of_dims = 1
+            for dim in memref_size[i + 1 :]:
+                product_of_dims *= dim
+            index_values.append(f"{product_of_dims} * {indices[i]}")
+        return " + ".join(index_values)
+
+    @print.register
+    def _(self, op: gpu.ModuleEndOp, out_stream: IO[str]):
+        # Nothing to print :)
+        pass
+
+    @print.register
+    def _(self, op: arith.Constant, out_stream: IO[str]):
+        value = int(str(op.attributes.get("value")).split()[0])
+        cons_type = op.result.type
+        if op.result.type.name == "index":
```

Comment:
I think we'd rather check the type than the name of the type (maybe using `isinstance`?)